### PR TITLE
convert rubygem-katello-nightly-release to groovy (baby)

### PR DESCRIPTION
**Pull request by komidore64**

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemKatelloNightlyRelease.groovy [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        timeout(time: 60, unit: 'MINUTES')
+        disableConcurrentBuilds()
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage ("Build Gem") {
+            steps {
+                dir('katello') {
+                    configureRVM('2.2')
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemKatelloNightlyRelease.groovy]: I generally like magic numbers to be in variables, especially if they're being reused :)

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemKatelloNightlyRelease.groovy [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                    git url: "https://github.com/Katello/katello", branch: 'master'
+                    withRVM(["gem build katello.gemspec"], '2.2')
+                    archiveArtifacts artifacts: '*.gem'
+                }
+            }
+        }
+
+        stage('Trigger RPM Build') {
+            steps {
+                build job: 'packaging_build_rpm', propagate: true, parameters: [
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemKatelloNightlyRelease.groovy]: I wonder if we could convert this job to a groovy step so you see everything inside a single job. We could reuse it in every nightly pipeline.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemKatelloNightlyRelease.groovy (hunk 1 above)
Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemKatelloNightlyRelease.groovy]: magic number has been variablized

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemKatelloNightlyRelease.groovy (hunk 2 above)
Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemKatelloNightlyRelease.groovy]: I would like to do that, but I think that'd outside the scope of this pull request. I will make a note that this is what we want to do down the road.

**Pull request merged**